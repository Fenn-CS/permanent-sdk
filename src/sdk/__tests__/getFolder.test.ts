import nock from 'nock';
import { getFolder } from '..';

describe('getFolder', () => {
  it('should return a Folder', async () => {
    nock('https://permanent.local')
      .get('/api/folder/getWithChildren')
      .query({
        folderId: 7457,
        archiveId: 1,
      })
      .replyWithFile(
        200,
        `${__dirname}/fixtures/getFolder/folder.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const folder = await getFolder(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      7457,
      1,
    );

    expect(folder).toMatchSnapshot();
  });
});
