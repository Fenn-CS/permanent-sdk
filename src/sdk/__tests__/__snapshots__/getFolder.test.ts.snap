// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFolder should return a Folder 1`] = `
Object {
  "archiveRecords": Array [
    Object {
      "createdAt": 2022-07-19T15:26:31.000Z,
      "displayDate": 2022-06-01T18:44:43.000Z,
      "displayName": "face-in-clouds",
      "fileSystemCompatibleName": "face-in-clouds.gif",
      "files": Array [],
      "id": 512,
      "status": "status.generic.ok",
      "type": "type.record.image",
      "updatedAt": 2022-07-19T15:30:23.000Z,
    },
  ],
  "createdAt": 2022-07-19T14:00:45.000Z,
  "displayDate": 2022-06-01T18:44:43.000Z,
  "fileSystemCompatibleName": "testing",
  "folders": Array [
    Object {
      "archiveRecords": Array [],
      "createdAt": 2022-07-19T15:26:40.000Z,
      "displayDate": 2022-07-19T15:26:40.000Z,
      "fileSystemCompatibleName": "Internal folder",
      "folders": Array [],
      "id": 7458,
      "name": "Internal folder",
      "size": 0,
      "updatedAt": 2022-07-19T15:27:06.000Z,
    },
  ],
  "id": 7457,
  "name": "testing",
  "size": 0,
  "updatedAt": 2022-07-19T15:30:22.000Z,
}
`;
